### DISCUSSION TITLE
Prevent problems with ACFML's "sync translations" option

### DISCUSSION THREAD

**Discussion by aaronmeder** [flyntwp/flynt#441] [General]
Hey everyone

This is actually not a problem of Flynt itself but rather an issue that affects Flynt based sites because it uses a flexible content field as the main content field of a pages/posts - so I thought it's worth sharing because some of you might also be affected by it.

## Problem
We recently ran into trouble on some websites that use Flynt and are made multilingual by using WPML while also having WPML's Advanced Custom Fields Multilingual plugin activated. There is a "Synchronise translations: synchronise repeater and flexible sub-fields positions in post translations (record drag-and-drop moves and do the same moves in other translations) " below flexible content fields (so below the Page components on pages):

<img width="965" alt="Bildschirmfoto 2022-05-19 um 17 46 52" src="https://user-images.githubusercontent.com/5564163/169478788-f765894a-a9b8-4290-834f-f017afbfd25f.png">

This can make a mess if your pages have different content in the translated languages.

## Solution

👉🏻 I've found and documented a way to disable the option by default:
[How to disable the “Synchronise repeater and flexible sub-fields positions in post translations” WPML option](https://telltec.ch/how-to-disable-the-synchronise-repeater-and-flexible-sub-fields-positions-in-post-translations-wpml-acf-option/)

The option / checkbox only recently came to our attention but it seems to have been there for a while.

Hope it helps!
Aaron

**Comment by steffenbew** [comment]
Thanks for sharing, @aaronmeder!

**Comment by timohubois** [comment]
@aaronmeder many thanks for sharing this – it helped me today and I had a similar topic today and created a drop-in for the `inc` folder which may can help in specific cases.

It does the following:
1. Define a constant that sets the default value for the repeater sync checkbox.
2. Remove the meta box that displays the repeater sync checkbox in the post edit screen.
3. Delete the option that stores the sync status in the database.

And can be found here: https://github.com/awesome-flynt/flynt-add-ons/blob/main/inc/acmlDisableRepeaterSync.php

**Reply by aaronmeder** [reply]
I think this completes fixing the whole impact of this function very well. Thanks a lot 💯 

Could this even be an idea for core Flynt? Because the repeater sync functionality will most likely hurt any site that doesn't want to do only 1:1 copies in other languages. At least for us at Soda disabling the repeater sync functionality would be something we do on every site.

**Reply by timohubois** [reply]
@aaronmeder glad to read from you. I like the idea to add it to the core, but think the core should stay be as lean as possible. 

While thinking more about this topic I shortly created a plugin (submitted it for approval to get added to official plugin repository as well): https://github.com/timohubois/disable-sync-repeater-option-for-acfml

**Reply by aaronmeder** [reply]
That is very helpful, thanks a lot. I think keeping the core lean is also great :)

**Comment by timohubois** [comment]
@aaronmeder the plugin got reviewed and is available now: https://wordpress.org/plugins/disable-sync-repeater-option-for-acfml/
